### ISSUE TITLE
Fn::Select not rejected as a bare value in object-typed lifecycle policies (CreationPolicy/UpdatePolicy)

### ISSUE THREAD

**Issue by kddejong** [aws-cloudformation/cfn-lint#4645]
### Summary

`Fn::Select` is not rejected when used as the whole value of an object-typed
resource attribute such as `CreationPolicy` / `UpdatePolicy`, even though
CloudFormation rejects it there (`Template format error: Expected an object`)
unless it is resolved by a transform.

This is a known follow-up carved out of #4641, which fixed the equivalent cases
for `Ref`, `Fn::Sub`, and `Fn::FindInMap`.

### Example (no transform) — not flagged today, but CloudFormation rejects it

```yaml
Resources:
  WaitCondition:
    Type: AWS::CloudFormation::WaitCondition
    CreationPolicy: !Select [0, [{ResourceSignal: {Count: 0, Timeout: PT1M}}]]
```

Real CloudFormation `CreateStack` → `Template format error: Expected an object,
received {"Fn::Select": ...}`. `cfn-lint` reports nothing.

### Why it's not caught

Unlike `Fn::Sub` (output type `string`) and `Fn::FindInMap` (output type
scalar/array), `Fn::Select` declares its output type as `all_types` (in
`src/cfnlint/rules/functions/Select.py`), which includes `object`. So
`validate_fn_output_types` considers it compatible with a `type: object` slot
and no error is produced.

`all_types` is arguably correct for `Fn::Select` in general: it *can* legitimately
resolve to an object, e.g. selecting from a list of objects produced by
`Fn::FindInMap` — but only **with** `AWS::LanguageExtensions`, which resolves
the `Fn::Select`/`Fn::FindInMap` pre-deploy:

```yaml
Transform: AWS::LanguageExtensions
Mappings:
  M:
    k:
      v:
        - ResourceSignal: {Count: 0, Timeout: PT1M}
Resources:
  WaitCondition:
    Type: AWS::CloudFormation::WaitCondition
    CreationPolicy: !Select [0, !FindInMap [M, k, v]]   # CREATE_COMPLETE with the transform
```

So this can't be fixed simply by narrowing `Fn::Select`'s output type. It needs
a mechanism that distinguishes "structurally allowed at this slot" (a per-location
function allow-list) from "could resolve to this type" — and ideally accounts for
whether `AWS::LanguageExtensions` is present.

### Notes / options

- A per-location allow-list (only object-literal or `Fn::If` at these lifecycle
  policy slots) would catch `Fn::Select` too, but that pattern was intentionally
  avoided in #4641 in favor of the type/output-type approach.
- Behavior verified against live CloudFormation.
